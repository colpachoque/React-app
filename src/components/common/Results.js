import React from "react";
import styles from "../header/Header.css";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import Radio from "@material-ui/core/Radio";
import { updateSortBy } from "../../actions/search.action";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

const Results = props => (
  <div className={styles.results}>
    <div className={styles.resultsContainer}>
      {!props.location.pathname.includes("/movie/") ? (
        <>
          <div className={styles.resultsInfo}>{props.info}</div>
          <div className={styles.resultsSort}>Sort by</div>
          <div className={styles.radioGroup}>
            <RadioGroup
              row
              value={props.sortBy}
              onChange={({ target }) => {
                props.updateSortBy(target.value);
              }}
            >
              <FormControlLabel
                value="release-date"
                control={<Radio color="primary" />}
                label="Release Date"
                labelPlacement="start"
              />
              <FormControlLabel
                value="rating"
                control={<Radio color="primary" />}
                label="Rating"
                labelPlacement="start"
              />
            </RadioGroup>
          </div>
        </>
      ) : (
        <div className={styles.resultsInfo}>Films by {props.genre}</div>
      )}
    </div>
  </div>
);

const mapStateToProps = state => {
  return { sortBy: state.search.sortBy, genre: state.movies.genre };
};
export default withRouter(
  connect(
    mapStateToProps,
    { updateSortBy }
  )(Results)
);
