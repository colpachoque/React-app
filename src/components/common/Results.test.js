import React from "react";
import Results from "./Results";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
describe("Results component", () => {
  it("should be render correctly", () => {
    const component = shallow(<Results />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
