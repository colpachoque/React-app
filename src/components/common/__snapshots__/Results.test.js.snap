// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results component should be render correctly 1`] = `
<div
  className="results"
>
  <div
    className="resultsContainer"
  >
    <div
      className="resultsInfo"
    />
    <div
      className="resultsSort"
    >
      Sort by
    </div>
  </div>
</div>
`;
